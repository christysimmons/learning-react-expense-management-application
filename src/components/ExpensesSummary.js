import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from "../selectors/expenses-total";

/*
---First shot---

const ExpensesSummary = (props) => (
            <div>
            <p> 
                Viewing&nbsp;
                {props.expenses.length} 
                {(props.expenses.length === 1 ) ? " expense " : " expenses " }
                totaling&nbsp;
                {numeral(getExpensesTotal(props.expenses) / 100 ).format('$0,0.00')}.
            </p>
            </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};*/

//Second shot

export const ExpensesSummary = ({ expensesTotal, expenseCount }) => {
    const formatExpenseTotal = numeral(expensesTotal / 100 ).format('$0,0.00')
    return (
        <div className="page-header">
            <div className="content-container">  
                <h2 className="page-header__title">
                Viewing <span>{expenseCount}</span> {(expenseCount === 1 ) ? "expense" : "expenses" } totalling <span>{formatExpenseTotal}</span>.
                </h2>
                <div className="page-header__actions">
                <Link to="/create" className="button">Add a new expense </Link> 
                </div>
            </div>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    }

}

export default connect(mapStateToProps)(ExpensesSummary);