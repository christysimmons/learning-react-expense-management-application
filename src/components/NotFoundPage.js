import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        <p>404 not found. <Link to="/"> Go home instead?</Link></p>
    </div>
)

export default NotFoundPage;