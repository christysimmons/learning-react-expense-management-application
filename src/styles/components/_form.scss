.form {
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: $sm-size;
    }
    > button {
        width: 8.6rem;
    }
}

.form__error {
    margin: 0 0 $med-size 0;
    font-style: italic;
}