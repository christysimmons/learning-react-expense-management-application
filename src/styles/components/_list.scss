.list-header {
    background-color: $off-white;
    border: 1px solid darken($off-white, 7%);
    color: $gray;
    display: flex;
    justify-content: space-between;
    padding: $sm-size $med-size;
}

.list-body {
    margin-bottom: $med-size;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $lg-size;
    }
}

.list-item {
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $darkgray;
    display: flex;
    flex-direction: column;
    padding: $sm-size;
    text-decoration: none;
    transition: background .3s ease;
    &:hover {
        background: $off-white;
    }
    @media (min-width: $desktop-breakpoint) {
        align-items: center;
        flex-direction: row;
        justify-content: space-between;
        padding: $med-size;
    }
}

.list-item--message {
    align-items: center;
    justify-content: center;
    color: $gray;
    padding: $med-size;
    &:hover {
        background: none;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__subtitle {
    color: $gray;
    font-size: $font-size-small;
}

.list-item__amount {
    margin: $sm-size 0 0 0;
    @media (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $sm-size;
    }
}