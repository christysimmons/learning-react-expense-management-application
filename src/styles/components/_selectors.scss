.button {
    background-color: $blue;
    border: none;
    border-radius: 2px;
    box-shadow: none;
    -moz-box-shadow: none;
    -webkit-box-shadow: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    outline: 0;
    padding: $sm-size;
    text-decoration: none;
}

.button--link {
    background-color: rgba(0,0,0,0);
}

.button--secondary {
    background-color: #888888;
}