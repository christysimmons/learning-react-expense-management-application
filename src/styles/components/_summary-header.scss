.page-header {
    background-color: $off-white;
    margin-bottom: $lg-size;
    padding: $lg-size 0;
}

.page-header__actions {
    margin-top: $med-size;
}

.page-header__title {
    font-weight: 300;
    margin: 0;
    span { 
        font-weight: bold;
    }
}