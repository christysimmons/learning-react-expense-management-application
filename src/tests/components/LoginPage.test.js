import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should load LoginPage correctly', () => {
    //Method for enzyme
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
    
});

test('should have logged in on click', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin}/>);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});