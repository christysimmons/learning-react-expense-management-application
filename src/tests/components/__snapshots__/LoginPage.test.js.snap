// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load LoginPage correctly 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
